src/net/fd_wasi.go

```diff
@@ -0,0 +1,78 @@
+//go:build wasi
+// +build wasi
+
+package net
+
+import (
+	"os"
+	"syscall"
+	"time"
+)
+
+// Network file descriptor.
+type netFD struct {
+	*os.File
+
+	net   string
+	laddr Addr
+	raddr Addr
+}
+
+// Read implements the Conn Read method.
+func (c *netFD) Read(b []byte) (int, error) {
+	// TODO: Handle EAGAIN and perform poll
```

Comment:
```suggestion
	// TODO: Handle EAGAIN and perform poll similar to poll/fd_unix.go
```

Reply:
Well, not necessarily same as https://github.com/golang/go/blob/master/src/internal/poll/fd_unix.go
It's circling back to https://github.com/tinygo-org/tinygo/pull/2748/files#r979834379 in my head - is it a goal of this project to be as similar to upstream Go and minimize diff or not? A simpler and more minimal poll implementation could probably suit this project better, but I am not sure what's the preferred approach.

Reply:
I'll leave this one to the maintainers to decide. thanks for clarifying!

Reply:
To the style question, my impression is that the project tries in this order:
1. directly use upstream files (no copying needed)
2. borrow a subset of upstream files, minimizing diffs but only borrowing the lines that are needed
3. replace upstream files with something more suited to tinygo's use case (but still kind of minimizing diffs if possible)

Reply:
Upstream's networking is... tricky because of its interaction with goroutines, as you've noticed.
How useful is networking without getting that bit right?  
Will your code (e.g. netfd.Read()) pause all goroutines while blocking on networking? 
I haven't checked yet.